From b61cc8fd6d99de86dec9c91e8aa0ad4cb1c6f596 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Sun, 21 Nov 2021 18:09:01 +0200
Subject: [PATCH] [SW-62976] gaudi2: adjust fit response offset

Gauid2's FW, sends a wrong memory offset as part of the FWL protocol's
return status command.An accompany FIT response, may be sent back to the
FWL. It is accessed by sending the offset in SRAM to the FWL as part
of the status command.

Currently the preboot send wrong offset address to the FWL while
responding with the proto status command.

The same occurs with the FWD_MAP response.

Change-Id: I774210b741a62081c77f18e2759bc41af8892227
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/lib/preboot/module_fwupd.c b/lib/preboot/module_fwupd.c
index 8cbb4b4..67d0973 100644
--- a/lib/preboot/module_fwupd.c
+++ b/lib/preboot/module_fwupd.c
@@ -150,7 +150,7 @@
 					  enum hl_upd_sts sts, void *offset)
 {
 	struct hl_upd_status status;
-	uintptr_t off = (uintptr_t)offset;
+	uintptr_t address = (uintptr_t)offset;
 
 	if (state->use_security && offset)
 		preboot_security_dynamic_img_sram_open(state->mregion,
@@ -158,7 +158,7 @@
 
 	status.val = 0;
 	status.status = sts;
-	status.offset = (u32)(off & ~HL_UPD_STS_OFFSET_MASK);
+	status.offset = (u32)((address - SDRAM_BASE) & ~HL_UPD_STS_OFFSET_MASK);
 	state->sts_writel(status.val);
 }
 
